x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/SecurityIndexManager.java

```diff
@@ -214,6 +210,18 @@ public void removeStateListener(BiConsumer<State, State> listener) {
         stateChangeListeners.remove(listener);
     }
 
+    /**
+     * Get the minimum security index mapping version in the cluster
+     */
+    private SystemIndexDescriptor.MappingsVersion getMinSecurityIndexMappingVersion(ClusterState clusterState) {
+        var minClusterVersion = clusterState.getMinSystemIndexMappingVersions().get(systemIndexDescriptor.getPrimaryIndex());
+        // Can be null in mixed clusters
+        if (minClusterVersion == null) {
+            return systemIndexDescriptor.getMappingsVersion();
```

Comment:
Could you expand this comment a bit to explain why we want to return the current version in this case (as opposed to `null` or a zero version).

Reply:
Added some more context to the comment. 

If we return a min version that isn't valid here we'll end up trying to get a descriptor compatible with that invalid version (return 0 or null) when we try to update the mapping, so the mapping update fails (the descriptor is null). An option could be to not do an update but I think in this case you want to do an update because the cluster state doesn't any security index mapping version yet and should be updated.